src/Data/List/Relation/Binary/Permutation/Propositional.agda

```diff
@@ -30,21 +38,32 @@ open import Relation.Binary.Reasoning.Syntax
 infix 3 _↭_
 
 data _↭_ : Rel (List A) a where
-  refl  : ∀ {xs}        → xs ↭ xs
-  prep  : ∀ {xs ys} x   → xs ↭ ys → x ∷ xs ↭ x ∷ ys
-  swap  : ∀ {xs ys} x y → xs ↭ ys → x ∷ y ∷ xs ↭ y ∷ x ∷ ys
-  trans : ∀ {xs ys zs}  → xs ↭ ys → ys ↭ zs → xs ↭ zs
+  refl  : xs ↭ xs
+  prep  : ∀ x → xs ↭ ys → x ∷ xs ↭ x ∷ ys
+  swap  : ∀ x y → xs ↭ ys → x ∷ y ∷ xs ↭ y ∷ x ∷ ys
+  trans : xs ↭ ys → ys ↭ zs → xs ↭ zs
+
+-- Constructor aliases
+
+↭-refl : Reflexive _↭_
+↭-refl = refl
+
+↭-prep  : ∀ x → xs ↭ ys → x ∷ xs ↭ x ∷ ys
+↭-prep = prep
+
+↭-swap  : ∀ x y → xs ↭ ys → x ∷ y ∷ xs ↭ y ∷ x ∷ ys
+↭-swap = swap
```

Comment:
As always I am going to be annoying and complain that I don't like prefixed
names because users can either use the short names or import the module
qualified if they need to manually disambiguate.

The same goes for `split` being renamed to `↭-split`  (although in that case
the name `split` is already pretty poor IMO).

Reply:
@gallais I don't find the comment annoying!
What I find annoying is the the abstraction failure when explicitly exporting constructors from an inductive definition which should remain (externally, to clients) abstract, IMHO (cf. `z<s` in `Data.Nat.Base`, and `0<1+n` in `Data.Nat.Properties`, not that I/we are entirely consistent in enforcing this...)
Since this is also part of a greater project of refactoring `Permutation` towards completing #1761 / #1354 I wanted to make sure that in `Properties`, references to the constructors were abstracted on those lines before proceeding further.

As regards `↭-split`, this is *not* a renaming of `split`, but a significant generalisaiton which permits a much improved proof of `dropMiddleElement`... so a cognate name change is needed to relfect the 'bettre' typing!